File: app/src/main/java/com/jurtz/marcel/shoppinglist/model/ShoppingListItem.java
package com.jurtz.marcel.shoppinglist.model;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.PrimaryKey;

import static android.arch.persistence.room.ForeignKey.CASCADE;

@Entity(
        tableName = "shopping_list_item",
        foreignKeys = @ForeignKey(
                entity = ShoppingList.class,
                parentColumns = "id",
                childColumns = "list_id",
                onDelete=CASCADE))
public class ShoppingListItem {

    /*
    public ShoppingListItem(int id, String description, int timestampSeconds) {
        this.id = id;
        this.description = description;
        this.timestampSeconds = timestampSeconds;
    }
    */

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    public int id;

    @ColumnInfo(name = "list_id")
    public int listId;

    @ColumnInfo(name = "description")
    public String description;

    @ColumnInfo(name = "timestamp_seconds")
    public int timestampSeconds;
}

File: app/src/main/java/com/jurtz/marcel/shoppinglist/model/ShoppingListItem.kt
package com.jurtz.marcel.shoppinglist.model

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import java.util.*

@Entity(tableName = "shopping_list_item")
data class ShoppingListItem(
        @ColumnInfo(name = "description")
        var description: String? = null,

        @ColumnInfo(name = "timestamp")
        var timestamp: Int = (Calendar.getInstance().timeInMillis / 1000).toInt()
) {

    @PrimaryKey(autoGenerate = true)
    private val id: Int = 0

}